lib/elixir/test/elixir/module/types/descr_test.exs

```diff
@@ -0,0 +1,74 @@
+Code.require_file("type_helper.exs", __DIR__)
+
+defmodule Module.Types.DescrTest do
+  use ExUnit.Case, async: true
+
+  import Module.Types.Descr
+
+  defp to_quoted_string(descr), do: Macro.to_string(to_quoted(descr))
+
+  describe "union" do
+    test "bitmap" do
+      assert union(integer(), float()) == union(float(), integer())
+    end
+
+    test "term" do
+      assert union(term(), float()) == term()
+      assert union(term(), binary()) == term()
+    end
+
+    test "none" do
+      assert union(none(), float()) == float()
+      assert union(none(), binary()) == binary()
+    end
+  end
+
+  describe "intersection" do
+    test "bitmap" do
+      assert intersection(integer(), union(integer(), float())) == integer()
+      assert intersection(integer(), float()) == none()
+    end
+
+    test "term" do
+      assert intersection(term(), float()) == float()
+      assert intersection(term(), binary()) == binary()
+    end
+
+    test "none" do
+      assert intersection(none(), float()) == none()
+      assert intersection(none(), binary()) == none()
+    end
+  end
+
+  describe "difference" do
+    test "bitmap" do
+      assert difference(float(), integer()) == float()
+      assert difference(union(float(), integer()), integer()) == float()
+      assert difference(union(float(), integer()), binary()) == union(float(), integer())
+    end
+
+    test "term" do
+      assert difference(float(), term()) == none()
+      assert difference(float(), term()) == none()
```

Comment:
double-checking, is this clause is duplicated on purpose?

Reply:
Good find, thank you!